pkg/virt-controller/watch/network.go

```diff
@@ -62,3 +62,20 @@ func applyDynamicIfaceRequestOnVMI(vm *v1.VirtualMachine, vmi *v1.VirtualMachine
 	}
 	return vmiSpecCopy
 }
+
+func clearDetachedInterfaces(vmiSpec *v1.VirtualMachineInstanceSpec, statusIfaces []v1.VirtualMachineInstanceNetworkInterface) {
+	indexedStatusIfaces := vmispec.IndexInterfacesFromStatus(statusIfaces,
+		func(ifaceStatus v1.VirtualMachineInstanceNetworkInterface) bool { return true })
+
+	var attachedIfaces []v1.Interface
+	for _, iface := range vmiSpec.Domain.Devices.Interfaces {
+		if _, existInStatus := indexedStatusIfaces[iface.Name]; !existInStatus &&
```

Comment:
- It will be nicer to give the direct condition for an interface to be left on the spec.
  (i.e. avoid using `continue`)

- The name `attachedIfaces` is confusing, at least to me.
   The scenario where an interface is in the spec, non-absent and missing from the status is special. It is not `attached` yet, but we are striving to see it plugged.

  Maybe use `ifaces` or `interfaces`, I do not have a good name either.

Reply:
> I can add a comment right next to ...

I do not think so... If you need to explain it, it means it needs to be simplified.

> The direct condition would be interfaces that exist in spec and not status (the
> suggested unit tests pass whether a spec interface has an absent state or not).
> Sounds right?
> Do you think there are other scenarios that should be covered?

The condition is that it exists in the spec (assured by passing through the spec interfaces), that it is not in the status and that it has **no** absent.

If the tests are not covering a state where where the it is in the spec with absent and it is not in the status, then you should add it. In this case it should not be removed.

Reply:
> The condition is that it exists in the spec (assured by passing through the spec interfaces), that it is not in the status and that it has no absent.

I don't think it's going to work with having both terms "not exist in status" & "not absent".
If it's not absent, it shouldn't be removed from the spec, isn't it?

I meant for a condition to build an interfaces list that should **remain** in the spec.
Anything else will require additional indexing/filtering (not that it's bad..)

> If the tests are not covering a state where the it is in the spec with absent and it is not in the 
status, then you should add it. In this case it should not be removed.

There is a test that covers this scenario `should be cleared from spec, given absent iface and no associated status iface`.

A condition that says "exist in spec and not in status" would also satisfy the tests.
I can't think of a case where an interface does not exist in status and should remain in spec, is there one?

Reply:
> I don't think it's going to work with having both terms "not exist in status" & "not absent".
> If it's not absent, it shouldn't be removed from the spec, isn't it?

I do not understand to what you refer to. The condition is what to keep, not what you remove.
This is the condition I described.

> A condition that says "exist in spec and not in status" would also satisfy the tests.
> I can't think of a case where an interface does not exist in status and should remain in spec, is there one?

Yes, there is. If it exists in the spec, it does not mean it is yet plugged. It may happen later, but not yet.
E.g. multus has not reported it yet (maybe the pod is not there yet) and obviously virt-handler has not yet processed it yet.

Reply:
I got you wrong at first, thanks for the discussion.

Fixed the condition and added more tests to cover all stages in a spec interface lifecycle.